### PR TITLE
fix: [PL-30628]: Made x-header-Id field in Aws Auth of Hashicorp vault optional

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
### Summary
Made the field Server Id Header in aws auth of haschicorp vault as optional.
Testing:
Tested the CRUD flow with this field included and verified its working. (In local)
Tested the CURD flow without this filed and verified its working. (in local)
Tested update from other auth method to aws auth and vice-versa.

#### Screenshots
Screenshot 2023-02-07 at 10 41 46 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by VikasMaddukuriHarness** [comment]
retrigger coverage

File: src/modules/35-connectors/components/CreateConnector/HashiCorpVault/views/VaultConfigForm.tsx [hunk 1]
```diff
@@ -122,19 +122,7 @@ const VaultConfigForm: React.FC<StepProps<StepDetailsProps> & ConnectorDetailsPr
...
-                if (
-                  (prevStepData?.spec as VaultConnectorDTO)?.accessType === HashiCorpVaultAccessTypes.AWS_IAM ||
-                  value?.name?.length > 0
-                ) {
-                  return true
-                }
-                return false
-              }
-            )
+            then: Yup.object()
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/components/CreateConnector/HashiCorpVault/views/VaultConfigForm.tsx:125]: why do we need this validation?

Comment (VikasMaddukuriHarness) [reply src/modules/35-connectors/components/CreateConnector/HashiCorpVault/views/VaultConfigForm.tsx:125]: This checks whether the previous step (if this is updation case) if it is Aws_IAm or not and if yes, we will forward that object. 
If we remove this also it is working.

**Comment by VikasMaddukuriHarness** [comment]
retrigger coverage

**Comment by VikasMaddukuriHarness** [comment]
retrigger cypress-pipeline

**Review by meenakshi-harness — APPROVED** [review]

File: src/modules/35-connectors/components/CreateConnector/HashiCorpVault/views/VaultConfigForm.tsx (hunk 1 above)
Comment (VikasMaddukuriHarness) [reply src/modules/35-connectors/components/CreateConnector/HashiCorpVault/views/VaultConfigForm.tsx]: As discussed, this validation is no longer needed. Removed this.

**Review by swarajcheguri-harness — APPROVED** [review]

**Pull request merged**